client/allocrunner/consul_hook.go

```diff
@@ -87,6 +101,9 @@ func (h *consulHook) Prerun() error {
 }
 
 func (h *consulHook) prepareConsulTokensForTask(task *structs.Task, tg *structs.TaskGroup, tokens map[string]map[string]string) error {
+	if task == nil {
+		return nil
+	}
```

Comment:
Can this ever actually be nil? It seems like it'd be a programming error if so, in which case I'd expect this to return an error. Probably better to just let it panic.

Reply:
I'd rather return an error than panic here? But yes, this could only be a programming error.